src/dal/entities/repository-entity.ts

```diff
@@ -0,0 +1,29 @@
+export interface RepositoryEntity {
+    id: string;
+    deleted: boolean;
+    createdBy?: string;
+    lastUpdatedBy?: string;
+    creationDate: string;
+    lastUpdateDate: string;
+    dataVersion: number;
+    realityId: number;
+    classification?: string;
+    secretGroups?: string[];
+}
+
+export function isRepositoryEntity(object: any): boolean {
+    const joi = require('joi');
+    const schema = joi.object().keys({
```

Comment:
Create this schema outside the function, no need to create it every time function called